File: index/index-list-direct-pagination.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 

  const listParams = {
    id: '66e9358a808d95368f6f7a7c',
    name: 'sdk-ref-index-create-07',
    engineOptions: ['conversation', 'visual'],
    engineFamily: 'marengo',
    page: 1,
    pageLimit: 5,
    sortBy: 'updated_at',
    sortOption: 'asc'
  };
  
  const indexes = await client.index.list(listParams);
  indexes.forEach(index => {
    console.log(`ID: ${index.id}`);
    console.log(`  Name: ${index.name}`);
    console.log("  Engines:");
    index.engines.forEach((engine, index) => {
      console.log(`    Engine ${index + 1}:`);
      console.log(`      Name: ${engine.name}`);
      console.log(`      Options: ${JSON.stringify(engine.options)}`);
    });
    console.log(`  Video count: ${index.videoCount}`);
    console.log(`  Total duration: ${index.totalDuration} seconds`);
    console.log(`  Created at: ${index.createdAt}`);
    if (index.updatedAt) {
      console.log(`  Updated at: ${index.updatedAt}`);
    }
  });
})();

File: index/index-list-direct-pagination.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))


indexes = client.index.list(
    id="",
    name="",
    # page=1,
    # page_limit=5,
    # engine_options=["visual", "conversation"],
    # engine_family="marengo",
    # sort_by = "updated_at",
    # sort_option="asc",
    # created_at="2024-09-17T07:53:46.365Z",
    # updated_at="2024-09-17T07:53:46.365Z"
)
for index in indexes:
    print(f"ID: {index.id}")
    print(f"  Name: {index.name}")
    print("  Engines:")
    for i, engine in enumerate(index.engines, 1):
        print(f"    Engine {i}:")
        print(f"      Name: {engine.name}")
        print(f"      Options: {engine.options}")
    print(f"  Video count: {index.video_count}")
    print(f"  Total duration: {index.total_duration} seconds")
    print(f"  Created at: {index.created_at}")
    if index.updated_at:
        print(f"  Updated at: {index.updated_at}")
